pywbem/mof_compiler.py

```diff
@@ -599,42 +599,55 @@ def p_mp_createClass(p):
-                            continue
-                        # else obj.type is 'string'
-                        try:
-                            embedded_inst = obj.qualifiers['embeddedinstance']
-                        except KeyError:
+                            obj_ref = obj.reference_class.lower()
+                            if obj_ref not in dep_classes:
+                                dep_classes.append(obj_ref)
+                                dep_rslts.append((obj_ref, ce))
+                        elif obj.type == 'string':
+                            try:
+                                embedded_inst = \
+                                    obj.qualifiers['embeddedinstance']
+                            except KeyError:
+                                continue
+                            embedded_inst_cln = embedded_inst.value.lower()
+                            if embedded_inst_cln not in dep_classes:
+                                dep_classes.append(embedded_inst_cln)
+                                dep_rslts.append((embedded_inst_cln, ce))
+
                             continue
-                        embedded_inst = embedded_inst.value.lower()
-                        if embedded_inst not in dep_classes:
-                            dep_classes.append(embedded_inst)
-                        continue
-                    for klass in dep_classes:
-                        if klass in p.parser.classnames[ns]:
+                    for cln, err in dep_rslts:
+                        if cln in p.parser.classnames[ns]:
                             continue
                         try:
                             # don't limit it with LocalOnly=True,
                             # PropertyList, IncludeQualifiers=False, ...
                             # because of caching in case we're using the
                             # special WBEMConnection subclass used for
                             # removing schema elements
-                            p.parser.handle.GetClass(klass,
+                            p.parser.handle.GetClass(cln,
                                                      LocalOnly=False,
                                                      IncludeQualifiers=True)
-                            p.parser.classnames[ns].append(klass)
+                            p.parser.classnames[ns].append(cln)
                         except CIMError:
-                            moffile = p.parser.mofcomp.find_mof(klass)
+                            moffile = p.parser.mofcomp.find_mof(cln)
                             if not moffile:
-                                raise
-                            p.parser.mofcomp.compile_file(moffile, ns)
-                            p.parser.classnames[ns].append(klass)
+                                raise err
+                            try:
+                                p.parser.mofcomp.compile_file(moffile, ns)
```

Comment:
Hmm. I thought this function is just supposed to raise an error, and the compile is done elsewhere?